### PR TITLE
[Feature] support prune flat json on plan

### PR DESCRIPTION
**Pull request by Seaven**
Why I'm doing:
After flatten the JSON on BE's storage,  FE plan is needed to determine which subcolumns to select

What I'm doing:
for special JSON functions with JSON_PATH (e.g. get_json_xx, json_length....), supports using JSON_PATH to prune the flat-sub-column of JSON

This PR only includes changes on the FE Plan; and the flat JSON implementation on the BE is in another PR https://github.com/StarRocks/starrocks/pull/37577

* reuse semi-type's framework
* to compute the ColumnAccessPath base on json functions, BE will use the ColumnAccessPath to select the flat-sub-column of JSON 

```
MySQL x11> explain verbose select get_json_int(j1, '$.c') from js1;
+------------------------------------------------------------------------------------------------------------------------------------+
| Explain String                                                                                                                     |
+------------------------------------------------------------------------------------------------------------------------------------+
| RESOURCE GROUP: default_wg                                                                                                         |
|                                                                                                                                    |
| PLAN FRAGMENT 0(F01)                                                                                                               |
|   Output Exprs:5: get_json_int                                                                                                     |
|   Input Partition: UNPARTITIONED                                                                                                   |
|   RESULT SINK                                                                                                                      |
|                                                                                                                                    |
|   2:EXCHANGE                                                                                                                       |
|      cardinality: 1                                                                                                                |
|                                                                                                                                    |
| PLAN FRAGMENT 1(F00)                                                                                                               |
|                                                                                                                                    |
|   Input Partition: RANDOM                                                                                                          |
|   OutPut Partition: UNPARTITIONED                                                                                                  |
|   OutPut Exchange Id: 02                                                                                                           |
|                                                                                                                                    |
|   1:Project                                                                                                                        |
|   |  output columns:                                                                                                               |
|   |  5 <-> get_json_int[([3: j1, JSON, true], '$.c'); args: JSON,VARCHAR; result: INT; args nullable: true; result nullable: true] |
|   |  cardinality: 1                                                                                                                |
|   |                                                                                                                                |
|   0:OlapScanNode                                                                                                                   |
|      table: js1, rollup: js1                                                                                                       |
|      preAggregation: on                                                                                                            |
|      partitionsRatio=1/1, tabletsRatio=10/10                                                                                       |
|      tabletList=7477510,7477512,7477514,7477516,7477518,7477520,7477522,7477524,7477526,7477528                                    |
|      actualRows=1, avgRowSize=1025.0                                                                                               |
|      ColumnAccessPath: [/j1/c]                                                                                                     |
|      cardinality: 1                                                                                                                |
+------------------------------------------------------------------------------------------------------------------------------------+
29 rows in set
Time: 0.014s

```

## What type of PR is this:

- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: gensrc/proto/segment.proto [hunk 1]
```diff
@@ -190,6 +190,8 @@ message ColumnMetaPB {
     optional uint64 total_mem_footprint = 31;
     // for json column only
     optional JsonMetaPB json_meta = 32;
+    // for json flat column only
+    optional bytes name = 33;
```
Comment (mofeiatwork) [inline gensrc/proto/segment.proto:194]: why not put it into `json_meta`

Comment (Seaven) [reply gensrc/proto/segment.proto:194]: the column type after flattening JSON may not be JSON type in the future, so I wasn't put it into `json_meta`, but it's a good idea

Comment (Seaven) [reply gensrc/proto/segment.proto:194]: I remove the change in this PR, and reconsider it in next BE's PR

File: fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java [hunk 2]
```diff
@@ -297,6 +297,7 @@ public class SessionVariable implements Serializable, Writable, Cloneable {
     public static final String CBO_PUSH_DOWN_AGGREGATE = "cbo_push_down_aggregate";
     public static final String CBO_DEBUG_ALIVE_BACKEND_NUMBER = "cbo_debug_alive_backend_number";
     public static final String CBO_PRUNE_SUBFIELD = "cbo_prune_subfield";
+    public static final String CBO_PRUNE_JSON_SUBFIELD = "cbo_prune_json_subfield";
```
Comment (fzhedu) [inline fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java:300]: just use `cbo_prune_subfield` ok?

Comment (Seaven) [reply fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java:300]: Okkk

**Pull request closed without merging**